Prefix files with hyphens in path crash build

Because the matcher pattern on L245 identifies argument values by 'inverse-hyphen' class, paths are interpreted as terminating early.  Instead, argument values should continue until whitespace is found.

Example failure when there is a hyphen:
```
FAILURE: Build failed with an exception.

* Where:
Script '/Users/XX/clean/YY-ZZ/third_party/j2objc-gradle/j2objc.gradle' line: 245

* What went wrong:
Execution failed for task ':ZZ:j2objcTest'.
> java.io.FileNotFoundException: /Users/XX/clean/YY (No such file or directory)
```

PR incoming.

--- a/j2objc.gradle
+++ b/j2objc.gradle
@@ -232,7 +232,7 @@ class J2objcUtils {
     //   --prefixes dir/prefixes.properties --prefix com.ex.dir=Short --prefix com.ex.dir2=Short2
     static def prefixProperties(Project proj) {
         Properties props = new Properties()
-        def matcher = (proj.j2objcConfig.translateFlags =~ /--prefix(|es) +([^-]+)/)
+        def matcher = (proj.j2objcConfig.translateFlags =~ /--prefix(|es)\s+(\S+)/)
         def start = 0
         while (matcher.find(start)) {
             start = matcher.end()

